test/integration/BCoWHelper.t.sol

```diff
@@ -0,0 +1,229 @@
+    helper = new BCoWHelper(address(ammFactory));
+
+    deal(address(DAI), lp, 2 * VALID_AMOUNT);
+    deal(address(WETH), lp, 2 * VALID_AMOUNT);
+
+    vm.startPrank(lp);
+    weightedPool = ammFactory.newBPool();
+    basicPool = ammFactory.newBPool();
+
+    DAI.approve(address(weightedPool), type(uint256).max);
+    WETH.approve(address(weightedPool), type(uint256).max);
+    weightedPool.bind(address(DAI), VALID_AMOUNT, 8e18); // 80% weight
+    weightedPool.bind(address(WETH), VALID_AMOUNT, 2e18); // 20% weight
+
+    DAI.approve(address(basicPool), type(uint256).max);
+    WETH.approve(address(basicPool), type(uint256).max);
+    basicPool.bind(address(DAI), VALID_AMOUNT, 4.2e18); // no weight
+    basicPool.bind(address(WETH), VALID_AMOUNT, 4.2e18); // no weight
+
+    // finalize
+    weightedPool.finalize();
+    basicPool.finalize();
+
+    vm.stopPrank();
+  }
+
+  // NOTE: 1 ETH = 1000 DAI
+  uint256 constant INITIAL_SPOT_PRICE = 0.001e18;
+
+  function testBasicOrder() public {
+    IBCoWPool pool = IBCoWPool(address(basicPool));
+
+    uint256 ammWethInitialBalance = 1 ether;
+    uint256 ammDaiInitialBalance = 1000 ether;
+
+    deal(address(WETH), address(pool), ammWethInitialBalance);
+    deal(address(DAI), address(pool), ammDaiInitialBalance);
+
+    uint256 spotPrice = pool.getSpotPriceSansFee(address(WETH), address(DAI));
+    assertEq(spotPrice, INITIAL_SPOT_PRICE);
+
+    _executeHelperOrder(pool, ammWethInitialBalance, ammDaiInitialBalance);
+
+    uint256 postSpotPrice = pool.getSpotPriceSansFee(address(WETH), address(DAI));
+    assertEq(postSpotPrice, 1_052_631_578_947_368);
+  }
+
+  // NOTE: this test checks that the order generated by the helper is valid
+  function testValidOrder(uint256 balanceToken0, uint256 balanceToken1, uint256 priceSkewness) public {
+    IBCoWPool pool = IBCoWPool(address(basicPool));
+    balanceToken0 = bound(balanceToken0, 1e18, 1e36);
+    balanceToken1 = bound(balanceToken1, 1e18, 1e36);
+    // skew the price by max 50% (more could result in reverts bc of max swap ratio)
+    priceSkewness = bound(priceSkewness, 5000, 15_000);
+    vm.assume(priceSkewness != 10_000); // avoids no-skewness revert
+
+    vm.mockCall(
+      address(DAI), abi.encodeWithSelector(IERC20.balanceOf.selector, address(pool)), abi.encode(balanceToken0)
+    );
+    vm.mockCall(
+      address(WETH), abi.encodeWithSelector(IERC20.balanceOf.selector, address(pool)), abi.encode(balanceToken1)
+    );
+
+    uint256[] memory prices = new uint256[](2);
+    prices[0] = balanceToken1;
+    prices[1] = balanceToken0 * priceSkewness / 10_000;
+
+    // the helper generates the AMM order
+    (GPv2Order.Data memory ammOrder,,,) = helper.order(address(pool), prices);
+
+    if (priceSkewness > 10_000) {
+      assertEq(address(ammOrder.buyToken), address(DAI));
+    } else {
+      assertEq(address(ammOrder.buyToken), address(WETH));
+    }
+
+    // it should not revert
+    pool.verify(ammOrder);
```

Comment:
could we also test the signature with `isValidSignature`?

Reply:
chose `verify` because `isValidSignature` actually does check everything at the same time, while verify only checks for the validity of the order